src/api/app/models/project.rb

```diff
@@ -58,7 +58,10 @@ def autocomplete(search)
   has_many :messages, :as => :db_object, :dependent => :delete_all
   has_many :watched_projects, :dependent => :destroy, inverse_of: :project
 
-  has_many :linkedprojects, -> { order(:position) }, :class_name => 'LinkedProject', foreign_key: :db_project_id, :dependent => :delete_all
+  has_many :linked_projects, -> { order(:position) }, foreign_key: :db_project_id, :dependent => :delete_all
+  has_many :projects_linked, through: :linked_projects, class_name: 'Project', source: :linked_db_project
+  has_many :linking_projects, -> { order(:position) }, :class_name => 'LinkedProject', foreign_key: :linked_db_project_id, :dependent => :delete_all
+  has_many :projects_linking, through: :linking_projects, class_name: 'Project', source: :project
```

Comment:
the names in your proposal sound much better indeed.

However, I am wondering when this aliases are usable at all. Since it is important to order the links and handle also indirect links. At least when dealing with package objects inside.

So I wonder when one can use this aliases at all in real live.